lahja/asyncio/endpoint.py

```diff
@@ -192,11 +198,11 @@ def __init__(
                 async with self._received_response:
                     self._received_response.notify_all()
             elif isinstance(message, SubscriptionsUpdated):
-                self.subscribed_messages = message.subscriptions
                 async with self._received_subscription:
+                    self.subscribed_messages = message.subscriptions
                     self._received_subscription.notify_all()
-                if message.response_expected:
-                    await self.send_message(SubscriptionsAck())
+                    if message.response_expected:
```

Comment:
@lithp can you confirm this should be how this is structured or do you think it's ok to send the `Ack` after we've released the lock...

Reply:
I think it's fine to send the `Ack` from outside the lock's block, and maybe even preferable! The Condition is there to coordinate the local process, since the remote process doesn't even know about this Condition I don't think that releasing the lock first could create any race conditions.